File: 1127-last-stone-weight/last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq= new PriorityQueue<>((a,b)->b-a);

        for(int stone: stones ){
            pq.add(stone);
        }

        while(pq.size()>1){
            int first=pq.poll();
            int second=pq.poll();

            if(first!=second){
                pq.add(Math.abs(first-second));
            }
        }

        return pq.size()==1? pq.peek() :0 ;
    }
}

File: 1127-last-stone-weight/last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap=[]

        for stone in stones:
            heapq.heappush(heap, -1*stone)
        
        while len(heap)>1:
            first= heapq.heappop(heap)
            second=heapq.heappop(heap)
            if first!=second:
                heapq.heappush(heap,-1 * abs(abs(first)-abs(second)))
        
        return 0 if not heap else -1*heap[0]